test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/GeneratorUtil.java

```diff
@@ -0,0 +1,121 @@
+package uk.gov.justice.services.test.utils.core.random;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+
+/**
+ * 
+ * Utility class used by the email generator
+ * 
+ * @See EmailAddressGenerator
+ *
+ */
+public class GeneratorUtil {
+
+    /**
+     * Private constructor to prevent instantiation
+     */
+    private GeneratorUtil() {}
+
+    
+    public static final List<String> getTopLevelDomains(){
+        try {
+            final Properties props = new Properties();
+            props.load(GeneratorUtil.class.getClassLoader().getResourceAsStream("tlds.txt"));
+            return props.keySet().stream().map(o -> (String) o).collect(Collectors.toList());
+        } catch (IOException | NullPointerException e) {
+            // fall back to some tlds
+            return Arrays.asList(new String[]{"gov.uk", "co.uk", "org", "net", "com"});
+        }
+    }
+    
+
+    /**
+     * Generate a random string from an array of characters
+     * 
+     * @param validChars to chose from
+     * @param min length of the string
+     * @param max length of the string
+     * @return
+     */
+    public static String generateStringFromCharacters(final Random random, final char[] validChars, final int min,
+                    final int max) {
+        final int size = ThreadLocalRandom.current().nextInt(min, max + 1);
```

Comment:
Can we use the provided RANDOM instead of ThreadLocalRandom, as it would be useful to mock for testing.